### DISCUSSION TITLE
List objects are 1-based

### DISCUSSION THREAD

**Discussion by Arthurvdv** [StefanMaron/BusinessCentral.LinterCop#701] [New Rule]
Based on https://www.linkedin.com/posts/brad-prendergast_msdy365bc-businesscentral-activity-7216601980108984322-O42I

### Example
```AL
local procedure MyProcedure()
var
    myList: List of [Integer];
    i: Integer;
begin
    for i := 0 to myList.Count() do ;
end;

local procedure MyProcedure()
var
    myList: List of [Integer];
begin
    myList.Get(0);
end;
```

List objects are 1-based, so raise a diagnostic on possible cases of trying to retrieve the 0 index. Maybe even default as a warning?

Coincidentally today one of our internal PR's had this 😲

![image](https://github.com/user-attachments/assets/d05d1d11-ea4b-4f84-b48b-9fc05a261327)

**Comment by rvanbekkum** [comment]
Could be nice indeed, but you need to be very sure that it concerns an 'invalid' list access.
For `List.Get(i)` it might be much harder to detect whether it is actually an issue.

**Reply by Arthurvdv** [reply]
For now I'm going to start simple with these two scenario's

```AL
myList.Get(0);
```
```AL
for i := 0 to myList.Count() do;
```

You are right about the  'invalid' list access. and there are undoubtedly more complex possibilities, like the `List.Get(i)` you've mentioned. In these scenario's it's not always possible to determine the value of `i` for the code analyzer.
